src/Buildalyzer.Workspaces/AnalyzerResultExtensions.cs

```diff
@@ -267,17 +268,28 @@ private static CompilationOptions CreateCompilationOptions(IAnalyzerResult analy
             ?? Array.Empty<ProjectAnalyzer>();
 
         private static IEnumerable<DocumentInfo> GetDocuments(IAnalyzerResult analyzerResult, ProjectId projectId) =>
-            analyzerResult
-                .SourceFiles?.Where(File.Exists)
-                .Select(x => DocumentInfo.Create(
-                    DocumentId.CreateNewId(projectId),
-                    Path.GetFileName(x),
-                    loader: TextLoader.From(
-                        TextAndVersion.Create(
-                            SourceText.From(File.ReadAllText(x), Encoding.Unicode), VersionStamp.Create())),
-                    filePath: x))
+            GetDocuments(analyzerResult.SourceFiles, projectId);
+
+        private static IEnumerable<DocumentInfo> GetDocuments(IEnumerable<string> files, ProjectId projectId) =>
+           files?.Where(File.Exists)
+               .Select(x => DocumentInfo.Create(
+                   DocumentId.CreateNewId(projectId),
+                   Path.GetFileName(x),
+                   loader: TextLoader.From(
+                       TextAndVersion.Create(
+                           SourceText.From(File.ReadAllText(x), Encoding.Unicode), VersionStamp.Create())),
+                   filePath: x))
             ?? Array.Empty<DocumentInfo>();
 
+        private static IEnumerable<DocumentInfo> GetAdditionalDocuments(IAnalyzerResult analyzerResult, ProjectId projectId)
+        {
+            string projectDirectory = Path.GetDirectoryName(analyzerResult.ProjectFilePath);
+
+            return GetDocuments(

```

Comment:
You could probably reduce the number of null checks in the new code by falling back to the empty collection in this function. (just passing by)

Reply:
Nono, I didn't mean any dramatic changes here :)

I only meant you can do:
```
var additionalFiles = analyzerResult.AdditionalFiles ?? Array.Empty<DocumentInfo>;
return GetDocuments(additionalFiles.Select...
```
...and remove all the other question marks since it's all private code.

But that's all nitpicking.

Reply:
Done!

BTW, any ETA on an updated package?

Reply:
Cool! Well, I am not a maintainer, I am just following this repo :D 
@daveaglick should know.

Reply:
@daveaglick Any thoughts on this?

Reply:
Soon(ish)? I really wish I had more time for OSS maintainer duties lately - my email inbox is basically just hundreds of GitHub notifications I keep in there as a queue. And then every day I run out of time 😬.

I'll try to take a look at this PR and the new one for updating MS Build dependencies tomorrow morning before heading out for the weekend. Unless there are any red flags, I'll probably just merge both and bump the major version.

Anyone on this thread have an issue with dropping < .NET 6 support in a new major? Asking here since if this change gets bundled with that one in a single new version, you won't be able to get this change without bumping the major.

Reply:
It is not my call. That being said, the [package](https://github.com/Corniel/roslyn-test-tools) I use it for, is .NET 6 (and .NET 7).